backend/cabin-sorting/service/sortedHackers-service.ts

```diff
@@ -1,35 +1,171 @@
+    }
 
-      return Object.keys(applicationResponses).reduce(formatApplicationResponses, initialHackerData)
-    })
-    return formattedHackerData;
+    return Object.keys(applicationResponses).reduce(
+      formatApplicationResponses,
+      initialHackerData
+    );
+  });
+  return formattedHackerData;
 };
 
-const createSortedHacker = async (hacker : any) => {
-  const createResponse = await sortedHackersDao.createdSortedHacker(hacker)
+const createSortedHacker = async (hacker: any) => {
+  const createResponse = await sortedHackersDao.createdSortedHacker(hacker);
   return createResponse;
 };
-  
-export default {getSortedHackers, createSortedHacker};
\ No newline at end of file
+
+const getGroupedHackers = async () => {
+  const hackers = await assignHackerCabins();
+  console.log(hackers)
+  let cabinEmails: string[][] = [];
+
+  hackers.forEach((hacker) => {
+    console.log(hacker)
+  })
+
+  const groupedHackers = hackers.reduce((accum, currVal) => {
+    const assignedCabin = currVal.assignedCabin;
+    const cabinNum = cabinList.indexOf(assignedCabin);
+    if (cabinNum === -1) {
+      console.error(
+        `Cabin assigned to Hacker ${assignedCabin.id} could not be found`
+      );
+      return accum;
+    }
+    const hackerEmail = currVal.email;
+    if (!accum[cabinNum]) {
+      accum[cabinNum] = []
+    }
+    accum[cabinNum].push(hackerEmail);
+    
+    console.log(accum)
+    return accum;
+  }, cabinEmails);
+
+  return groupedHackers
+};
+
+function loadCSV(filepath: string, headers: boolean): any[] {
+  const csvFileAbsolutePath = path.resolve(
+    'cabin-sorting',
+    'data', 
+    'csv_inputs', 
+    filepath);
+  console.log(csvFileAbsolutePath)
+
+  // error handling in case file is missing
+  let fileContent;
+  try {
+    fileContent = fs.readFileSync(csvFileAbsolutePath, {
+      encoding: 'utf-8'
+    });
+  } catch (err) {
+    console.log(`File cannot be found: "${filepath}"`);
+    return [];
+  }
+
+  const options = {
+    delimiter: ',',
+    columns: headers
+  };
+  return parse(fileContent, options);
+}
+
+async function assignHackerCabins() {
+  hackerList = await getSortedHackers();
+  answerList = loadCSV('answer.csv', true);
+  cabinList = loadCSV('cabinTypes.csv', false)[0];
+
+  CABIN_SIZE = Object.keys(cabinList).length;
+  QUESTIONS_SIZE = Object.keys(answerList[0]).length;
+
+  // ensuring the CSV files exists before continuing
+  if (
+    hackerList.length === 0 ||
+    answerList.length === 0 ||
+    cabinList.length === 0
+  ) {
+    console.error(
+      'Please add the respective csv file(s) to the folder to run the sorting algorithm'
+    );
+    return [];
+  } else {
+    matchAnswers();
+    return hackerList;
+  }
+}
+
+function matchAnswers() {
+  hackerList.forEach((hacker: any) => {
+    // each element = a different cabin, all initialized to 0
+    const cabinScore = Array<number>(CABIN_SIZE).fill(0);
+
+    hydrateCabinScore(hacker, cabinScore);
```

Comment:
this is kind of a weird function name, could you make it something more descriptive